### PR TITLE
HPCC-16996 Show compiler errors and warnings generated by bundle install

### PR DESCRIPTION
**Pull request by richardkchapman**
Previous fix caused issues for thor PIPE command, if pipe->run is called
multiple times.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [x] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately

## Testing:
Thor, Roxie and hThor regression suite tests PIPE* run successfully (which I should have done last time).

Ran roxie versions of pipe*.ecl test under valgrind without issues. I don't know how to valgrind thor...

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: system/jlib/jthread.cpp [hunk 1]
```diff
@@ -1840,6 +1842,8 @@ protected: friend class PipeWriterThread;
         if (et)
         {
             et->stop();
+            if (clearStderr)
+                delete et;
             // NOTE - we don't delete it here, since we want to be able to still read the buffered data
```
Comment (ghalliday) [inline system/jlib/jthread.cpp]: slightly pedantic, but this comment should be moved down to the call from wait() - it is a bit confusing here.

**Comment by ghalliday** [comment]
@richardkchapman one minor comment.

**Comment by richardkchapman** [comment]
@ghalliday Updated code comments

**Pull request merged**